specifyweb/frontend/js_src/lib/components/FormSliders/IntegratedRecordSelector.tsx

```diff
@@ -54,6 +74,16 @@ function RecordSelectorFromCollection<SCHEMA extends AnySchema>({
     readonly relationship: Relationship;
     readonly defaultIndex?: number;
     readonly children: (state: RecordSelectorState<SCHEMA>) => JSX.Element;
+    readonly setRecordSetsPromise: React.Dispatch<
```

Comment:
Why is this branch not based on xml-editor?
If it was, then `setRecordSetsPromise` wouldn't even be needed as that branch removed the need for it